### PR TITLE
parse symphony.yml in ci command

### LINKED ISSUE #14: Parsing and reading symphony YAML
Initial work for CI and CD, we need to work off the `symphony.yaml` file

- [ ] Parsing the file
- [ ] Go data structure

### PR DESCRIPTION
**Pull request by sebastus**
resolves #14  

so the values in the config can be used in the program

--symphony-config can be aliased as --sc or -c

### CODE REVIEW

File: sample.symphony.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+environment: prod
```
Comment (benc-uk) [inline sample.symphony.yaml:1]: Maybe move this to an etc/ or samples/ folder in the repo? root is already getting cluttered?

File: cmd/ci.go [hunk 2]
```diff
@@ -8,20 +8,97 @@ package cmd
...
+	"io/ioutil"
+	"os"
 
+	"github.com/fatih/color"
 	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"gopkg.in/yaml.v2"
 )
 
+type SymphonyConfig struct {
```
Comment (benc-uk) [inline cmd/ci.go:20]: How about moving this into pkg/symphony ?
I expect a bunch of other things might need it down the line

File: cmd/ci.go [hunk 3]
```diff
@@ -8,20 +8,97 @@ package cmd
...
 	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Println("ci called")
+
+		symphonyConfigFileName, _ := cmd.Flags().GetString("symphony-config")
+		verbose, _ := cmd.Flags().GetBool("verbose")
+
+		err := readAndUnmarshallConfig(symphonyConfigFileName)
+		cobra.CheckErr(err)
+
+		if verbose {
```
Comment (benc-uk) [inline cmd/ci.go:55]: There's already a flag in pkg/console to handle this, so they can pass --debug to any command
And then in the code we call `console.Debug("this is only shown when debug is set!!")`

File: cmd/ci.go [hunk 4]
```diff
@@ -8,20 +8,97 @@ package cmd
...
+		}
+
+		run(symphonyConfigFileName)
 	},
 }
 
+func run(symphonyConfigFileName string) {
+	fmt.Println()
+
+	color.Red("Running CI command, config: %s", symphonyConfigFileName)
```
Comment (benc-uk) [inline cmd/ci.go:66]: Replace with console.Error or console.Info etc 
See pkg/console it's pretty simple

**Review by benc-uk — CHANGES_REQUESTED** [review]
Some feedback

File: sample.symphony.yaml (hunk 1 above)
Comment (sebastus) [reply sample.symphony.yaml:1]: done. also moved .rover.sample.yaml

File: cmd/ci.go (hunk 3 above)
Comment (sebastus) [reply cmd/ci.go:55]: done

File: cmd/ci.go (hunk 2 above)
Comment (sebastus) [reply cmd/ci.go:20]: done. also made it an OOP.

File: cmd/ci.go (hunk 4 above)
Comment (sebastus) [reply cmd/ci.go:66]: done

**Comment by sebastus** [comment]
strictly speaking, it's not necessary to parse symphony.yml in the top level ci command. but the ci commands are in another issue. So when this pr is merged and I get to the next one, I'll move the parse down to the sub-commands.

**Review by benc-uk — APPROVED** [review]

**Pull request merged**